llvm/test/CodeGen/BPF/BTF/map-def-nested.ll

```diff
@@ -0,0 +1,609 @@
+; CHECK-NEXT:        .ascii	"*const map_def::MyValue"       # string offset=77
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"MyValue"                       # string offset=101
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"*const [i32; 10]"              # string offset=109
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"*const [i32; 0]"               # string offset=126
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"HashMapDef<map_def::MyKey, map_def::MyValue, 10, 0>" # string offset=142
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"type"                          # string offset=194
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"key"                           # string offset=199
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"value"                         # string offset=203
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"max_entries"                   # string offset=209
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"map_flags"                     # string offset=221
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"UnsafeCell<map_def::HashMapDef<map_def::MyKey, map_def::MyValue, 10, 0>>" # string offset=231
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"HashMap<map_def::MyKey, map_def::MyValue, 10, 0>" # string offset=304
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"HASH_MAP"                      # string offset=353
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"&core::panic::panic_info::PanicInfo" # string offset=362
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"PanicInfo"                     # string offset=398
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"message"                       # string offset=408
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"location"                      # string offset=416
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"can_unwind"                    # string offset=425
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"force_no_backtrace"            # string offset=436
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"&core::fmt::Arguments"         # string offset=455
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"&core::panic::location::Location" # string offset=477
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"bool"                          # string offset=510
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"_info"                         # string offset=515
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"panic"                         # string offset=521
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	".text"                         # string offset=527
+; CHECK-NEXT:        .byte	0
+; CHECK-NEXT:        .ascii	"/home/vadorovsky/playground/map-def/map-def-ebpf/src/main.rs" # string offset=533
```

Comment:
various strings with paths from your machine have snuck in here. i'm not sure what the convention is for these tests, but in case it is to sanitize these, please do so.